### PR TITLE
Baz 7/filter

### PR DESCRIPTION
**Pull request by ShinK8245**
# BAZ-7/Filter

## Link to Story

https://re-boot.atlassian.net/browse/BAZ-8

Filters apply correctly to the product list, updating the displayed items based on selected criteria without page reloads.

Active filters are visually distinct, and users can easily identify which filters have been applied.

The clear or reset function works as expected, removing all selected filters and returning the product list to its default state.

The component's layout and functionality adapt well across devices, providing a consistent user experience.

The Filter Component meets accessibility standards, ensuring that all users can navigate and utilize the filters effectively.
Please include a summary of the change and which issue is fixed.

## Screenshot(s) / GIF(s):
Screen Shot 2024-04-18 at 11 59 45 AM
Screen Shot 2024-04-18 at 11 59 51 AM
Screen Shot 2024-04-18 at 11 59 19 AM

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Dependency / Configuration update
- [ ] Documentation update

## Checklist:

- [x] My code follows the project's style guidelines
- [x] I have created a Storybook story for my component(s)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,14 +39,14 @@
...
     "@chromatic-com/storybook": "^1.3.1",
-    "@storybook/addon-essentials": "^8.0.6",
-    "@storybook/addon-interactions": "^8.0.6",
-    "@storybook/addon-links": "^8.0.6",
-    "@storybook/addon-onboarding": "^8.0.6",
-    "@storybook/blocks": "^8.0.6",
-    "@storybook/nextjs": "^8.0.6",
-    "@storybook/react": "^8.0.6",
-    "@storybook/test": "^8.0.6",
+    "@storybook/addon-essentials": "^8.0.8",
```
Comment (Jack-Vu) [inline package.json:41]: need to rebase with dev or undo these changes, storybook dependencies shouldn't be changed

File: src/app/_components/Filter/Filter.tsx [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+function isColorFilterOption(
+  option: FilterOption,
+): option is ColorFilterOption {
+  return "color" in option;
+}
+
+const ChevronIcon: React.FC<ChevronIconProps> = ({ isOpen }) => {
+  return (
+    <ChevronDownIcon
+      style={{
```
Comment (Jack-Vu) [inline src/app/_components/Filter/Filter.tsx]: don't need the style={} you can add the styling straight into the className

File: src/app/_components/Filter/Filter.tsx [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+                      <div
+                        key={idx}
+                        className="mb-2 flex flex-col items-center justify-center hover:opacity-70"
+                      >
+                        {isColorFilterOption(option) && (
+                          <>
+                            {/* circled color options */}
+                            <div
+                              className={`flex h-6 w-6 cursor-pointer items-center justify-center rounded-full`}
+                              style={{
```
Comment (Jack-Vu) [inline src/app/_components/Filter/Filter.tsx:120]: here to shouldn't need style={} should be able to directly put it into the class name for tailwind

File: src/app/_components/Filter/Filter.tsx [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+  AdjustmentsHorizontalIcon,
+} from "@heroicons/react/24/outline";
+
+interface FilterOption {
+  value: string;
+  label: string;
+  checked: boolean;
+}
+
+interface ColorFilterOption extends FilterOption {
```
Comment (Jack-Vu) [inline src/app/_components/Filter/Filter.tsx:16]: a bit confused on why you need a second interface for color filter option you should be able to just pass it in the filter option and if it is optional you can make it optional and handle it that way

**Review by Jack-Vu — COMMENTED** [review]
Left a few comments

**Review by Muradil-Erkin — APPROVED** [review]
